subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationProcessor.java

```diff
@@ -36,17 +36,28 @@
 public class DefaultBuildOperationProcessor implements BuildOperationProcessor, Stoppable {
     private static final String LINE_SEPARATOR = SystemProperties.getInstance().getLineSeparator();
 
+    private final BuildOperationWorkerRegistry buildOperationWorkerRegistry;
     private final BuildOperationQueueFactory buildOperationQueueFactory;
     private final StoppableExecutor fixedSizePool;
 
-    public DefaultBuildOperationProcessor(BuildOperationQueueFactory buildOperationQueueFactory, ExecutorFactory executorFactory, int maxWorkerCount) {
+    public DefaultBuildOperationProcessor(BuildOperationWorkerRegistry buildOperationWorkerRegistry, BuildOperationQueueFactory buildOperationQueueFactory, ExecutorFactory executorFactory, int maxWorkerCount) {
+        this.buildOperationWorkerRegistry = buildOperationWorkerRegistry;
         this.buildOperationQueueFactory = buildOperationQueueFactory;
         this.fixedSizePool = executorFactory.create("build operations", maxWorkerCount);
     }
 
     @Override
     public <T extends BuildOperation> void run(BuildOperationWorker<T> worker, Action<BuildOperationQueue<T>> generator) {
-        BuildOperationQueue<T> queue = buildOperationQueueFactory.create(fixedSizePool, worker);
+        BuildOperationWorkerRegistry.Completion completion = buildOperationWorkerRegistry.maybeStartOperation();
```

Comment:
At this stage there isn't any `BuildOperationWorkerRegistry.Operation` registered during configuration. I thought I put a comment in about how this hack won't be necessary when we merge the operations in `BuildOperationExecuter` with those used in `BuildOperationWorkerRegistry`.